SYCL/ESIMD/lsc/Inputs/lsc_slm_load.hpp

```diff
@@ -0,0 +1,160 @@
+    static_assert(!transpose, "Conversion types may not use vector");
+    static_assert(VS == 1, "Only D32 and D64 support vector load");
+  }
+
+  static_assert(DS != lsc_data_size::u16u32h, "D16U32h not supported in HW");
+  static_assert(sizeof(T) >= 4,
+                "D8 and D16 are valid only in 2D block load/store");
+
+  uint16_t Size = Groups * Threads * VL * VS;
+
+  T vmask = (T)-1;
+  if constexpr (DS == lsc_data_size::u8u32)
+    vmask = (T)0xff;
+  if constexpr (DS == lsc_data_size::u16u32)
+    vmask = (T)0xffff;
+  if constexpr (DS == lsc_data_size::u16u32h)
+    vmask = (T)0xffff0000;
+
+  T old_val = get_rand<T>();
+  T new_val = get_rand<T>();
+
+  auto GPUSelector = gpu_selector{};
+  auto q = queue{GPUSelector};
+  auto dev = q.get_device();
+  std::cout << "Running case #" << case_num << " on "
+            << dev.get_info<info::device::name>() << "\n";
+  auto ctx = q.get_context();
+
+  // workgroups
+  cl::sycl::range<1> GlobalRange{Groups};
+  // threads in each group
+  cl::sycl::range<1> LocalRange{Threads};
+  cl::sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
+
+  T *out = static_cast<T *>(sycl::malloc_shared(Size * sizeof(T), dev, ctx));
+  for (int i = 0; i < Size; i++)
+    out[i] = 0;
+
+  std::vector<uint32_t> p(VL, 0);
+  if constexpr (!transpose)
+    for (int i = 0; i < VL; i++)
+      p[i] = (pmask >> i) & 1;
+
+  try {
+    buffer<uint32_t, 1> bufp(p.data(), p.size());
+
+    auto e = q.submit([&](handler &cgh) {
+      auto accp = bufp.template get_access<access::mode::read>(cgh);
+      cgh.parallel_for<KernelID<case_num>>(
+          Range, [=](cl::sycl::nd_item<1> ndi) SYCL_ESIMD_KERNEL {
+            constexpr uint16_t gran = 4;    // using oword write x1 to init SLM
+            constexpr uint16_t slm_blocks = // number of owords per Thread
+                VL * VS / gran + ((VL * VS) % gran ? 1 : 0);
+            constexpr uint16_t slm_ud_per_group = slm_blocks * Threads * gran;
+            constexpr uint16_t slm_size_per_group =
+                slm_ud_per_group * sizeof(T);
+            constexpr uint16_t slm_size = slm_size_per_group * Groups;
+
+            uint16_t globalID = ndi.get_global_id(0);
+            uint32_t elem_off = globalID * VL * VS;
+            uint32_t byte_off = elem_off * sizeof(T);
+
+            slm_init(slm_size);
+            if (ndi.get_local_id(0) == 0) {
+              uint32_t groupID = ndi.get_group(0);
+              uint32_t group_off = groupID * slm_size_per_group;
+              simd<T, gran> slm_val(new_val + groupID * slm_ud_per_group, 1);
+              for (int i = 0; i < slm_size_per_group; i += gran * sizeof(T)) {
+                slm_block_store<T, gran>(i + group_off, slm_val);
+                slm_val += gran;
+              }
+            }
+
+            barrier();
+
+            if constexpr (transpose) {
+              auto vals = lsc_slm_block_load<T, VS, DS, L1H, L3H>(byte_off);
```

Comment:
I think this will have to change - hints removed. Please see https://github.com/intel/llvm/pull/5512/files#r802905016

Reply:
Yes, it has been updated.